src/Agent.Worker/TestResults/TestResultsServer.cs

```diff
@@ -88,5 +98,24 @@ public void InitializeServer(VssConnection connection)
         {
             return await TestHttpClient.CreateTestSubResultAttachmentAsync(reqModel, projectName, testRunId, testCaseResultId, testSubResultId, cancellationToken);
         }
+
+        private const string EnablePublishToTcmServiceDirectlyFromTaskFF = "TestManagement.Server.EnablePublishToTcmServiceDirectlyFromTask";
+
+        private static bool GetFeatureFlagState(FeatureAvailabilityHttpClient featureAvailabilityHttpClient, string FFName)
+        {
+            try
+            {
+                var featureFlag = featureAvailabilityHttpClient?.GetFeatureFlagByNameAsync(FFName).Result;
```

Comment:
dont you want to cache it?

Reply:
I think this happens once per task execution or whenever TestRunPublisher is created. And this will not be used in future. As all the flows goes through PTR nuget library. So I don't want to do it.